File: Npc/11001673.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001673: Tinchai
/// </summary>
public class _11001673 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0711210007006723$
    // - Hah... I can barely breathe...
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0727223007006898$
                // - Now isn't a great time to talk!
                switch (selection) {
                    // $script:0727223007006899$
                    // - What are these things?
                    case 0:
                        return 40;
                }
                return -1;
            case (40, 0):
                // $script:0727223007006900$
                // - Whatever they are, they aren't good. Let's focus less on what they are and more on how to stop them.
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.SelectableDistractor,
            (40, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11001673.py
""" 11001673: Tinchai """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0711210007006723$
        # - Hah... I can barely breathe...
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0727223007006898$
        # - Now isn't a great time to talk!
        if pick == 0:
            # $script:0727223007006899$
            # - What are these things?
            return 40
        return -1

    def __40(self, pick: int) -> int:
        # $script:0727223007006900$
        # - Whatever they are, they aren't good. Let's focus less on what they are and more on how to stop them.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.SELECTABLE_DISTRACTOR
        elif (self.state, self.index) == (40, 0):
            return Option.CLOSE
        return Option.NONE
